pyqtgraph/functions.py

```diff
@@ -1794,9 +1800,47 @@ def arrayToQPath(x, y, connect='all'):
 
     ds = QtCore.QDataStream(buf)
     ds >> path
-
     return path
 
+def array2d_to_qpolygonf(xdata, ydata):
+    """
+    Utility function to convert two 1D-NumPy arrays representing curve data
+    (X-axis, Y-axis data) into a single polyline (QtGui.PolygonF object).
+    This feature is compatible with PyQt4, PyQt5 and PySide2 (requires QtPy).
+
+    License/copyright: MIT License © Pierre Raybaut 2020.
+
+    :param numpy.ndarray xdata: 1D-NumPy array (numpy.float64)
+    :param numpy.ndarray ydata: 1D-NumPy array (numpy.float64)
+    :return: Polyline
+    :rtype: QtGui.QPolygonF
+    """
+    if not (
+        xdata.size == ydata.size == xdata.shape[0] == ydata.shape[0]
+        and np.issubdtype(xdata.dtype, np.double)
+        and np.issubdtype(ydata.dtype, np.double)
+    ):
+        raise ValueError("Arguments must be 1D, float64 NumPy arrays with same size")
+    size = xdata.size
+    dtype = xdata.dtype
+    if QtVersion.startswith("5"):
+        polyline = QtGui.QPolygonF(size)
+    else:
+        polyline = QtGui.QPolygonF([QtCore.QPointF(xdata[0], ydata[0])] * size)
```

Comment:
For PySide6, you could do
```python
polyline = QtGui.QPolygonF()
polyline.resize(size)
```

Reply:
Love that suggestion, figured there was a "clean" way of doing that, constructing a list ahead of time seemed like a silly way of doing it.

Made the change and updated the PR.

Reply:
Looks like `polyline.resize` is not available in PyQt6 (╯°□°）╯︵ ┻━┻

Reply:
in case you're curious, here is a list of what's available to me:

```
>>> dir(polyline)
['__add__', '__class__', '__contains__', '__delattr__', '__delitem__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__iadd__', '__init__', '__init_subclass__', '__le__', '__len__', '__lshift__', '__lt__', '__module__', '__mul__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__rlshift__', '__rmul__', '__setattr__', '__setitem__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'append', 'at', 'boundingRect', 'clear', 'contains', 'containsPoint', 'count', 'data', 'fill', 'first', 'indexOf', 'insert', 'intersected', 'intersects', 'isClosed', 'isEmpty', 'last', 'lastIndexOf', 'mid', 'prepend', 'remove', 'replace', 'size', 'subtracted', 'swap', 'toPolygon', 'translate', 'translated', 'united', 'value']
```

Reply:
For PyQt6, you could use
```python
polyline = QtGui.QPolygonF()
polyline.fill(QtCore.QPointF(), size)
```

Reply:
that's _much_ nicer!

Reply:
much nicer solution; thanks for the suggestion.  Implemented now.